### PR TITLE
CSV e2e - wait for 2 pods ready before delete one pod

### LINKED ISSUE #2553: e2e "ClusterServiceVersion when a csv exists specifying two replicas with one max unavailable remains in phase Succeeded when only one pod is available" failure
## Bug Report
e2e test in CI failure.  This is listed in #2401.   
Recent happening: https://github.com/operator-framework/operator-lifecycle-manager/runs/4718817543?check_suite_focus=true#step:4:9568

```
ClusterServiceVersion when a csv exists specifying two replicas with one max unavailable
  remains in phase Succeeded when only one pod is available
  /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:504

[BeforeEach] ClusterServiceVersion
  /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:55
[BeforeEach] when a csv exists specifying two replicas with one max unavailable
  /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:413
[It] remains in phase Succeeded when only one pod is available
  /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:504
[AfterEach] ClusterServiceVersion
  /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:60
cleaning up ephemeral test resources...
deleting test subscriptions...
deleting test installplans...
deleting test catalogsources...
deleting test crds...
deleting test csvs...
test resources deleted


• Failure [4.748 seconds]
ClusterServiceVersion
/home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:43
  when a csv exists specifying two replicas with one max unavailable
  /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:404
    remains in phase Succeeded when only one pod is available [It]
    /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:504

    Failed after 1.049s.
    Expected
        <v1alpha1.ClusterServiceVersionPhase>: InstallReady
    to equal
        <v1alpha1.ClusterServiceVersionPhase>: Succeeded

    /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:513

    Full Stack Trace
    github.com/operator-framework/operator-lifecycle-manager/test/e2e.glob..func6.5.2()
        /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/csv_e2e_test.go:513 +0x4c3
    github.com/operator-framework/operator-lifecycle-manager/test/e2e.TestEndToEnd(0xc0007ce780)
        /home/runner/work/operator-lifecycle-manager/operator-lifecycle-manager/test/e2e/e2e_test.go:59 +0x2d6
    testing.tRunner(0xc0007ce780, 0x3580d80)
        /opt/hostedtoolcache/go/1.16.12/x64/src/testing/testing.go:1193 +0xef
    created by testing.(*T).Run
        /opt/hostedtoolcache/go/1.16.12/x64/src/testing/testing.go:1238 +0x2b3
```

**What did you do?**
CI e2e test failed 

**What did you expect to see?**
No failure

**What did you see instead? Under which circumstances?**
A clear and concise description of what you expected to happen (or insert a code snippet).

**Environment**
* operator-lifecycle-manager version:

* Kubernetes version information:

* Kubernetes cluster kind: 

**Possible Solution**

**Additional context**

### PR DESCRIPTION
**Pull request by akihikokuroda**
Signed-off-by: akihikokuroda <akihikokuroda2020@gmail.com>

**Description of the change:**
Make sure 2 pods are ready for the deployment before the test deletes one of them for testing.

**Motivation for the change:**
Closes #2553

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/doc`
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: test/e2e/csv_e2e_test.go [hunk 1]
```diff
@@ -502,6 +502,13 @@ var _ = Describe("ClusterServiceVersion", func() {
 		})
 
 		It("remains in phase Succeeded when only one pod is available", func() {
+			Eventually(func() bool {
+				dep, err := c.GetDeployment(testNamespace, "deployment")
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(dep).ShouldNot(BeNil())
```
Comment (timflannagan) [inline test/e2e/csv_e2e_test.go:508]: nit: can we handle this error and nil check directly (e.g. `if err != nil ...`) without using gomega assertions?

File: test/e2e/csv_e2e_test.go [hunk 2]
```diff
@@ -502,6 +502,13 @@ var _ = Describe("ClusterServiceVersion", func() {
...
+				return dep.Status.ReadyReplicas == 2
+			}).Should(BeTrue())
```
Comment (timflannagan) [inline test/e2e/csv_e2e_test.go:510]: How do you feel about returning an integer instead and handling the comparison outside of the inline function block, e.g. `}).Should(Equal(2))`?

File: test/e2e/csv_e2e_test.go (hunk 1 above)
Comment (akihikokuroda) [reply test/e2e/csv_e2e_test.go:508]: I'll change it.  Thanks.

File: test/e2e/csv_e2e_test.go (hunk 2 above)
Comment (akihikokuroda) [reply test/e2e/csv_e2e_test.go:510]: It seems better.  I'll change it.  Thanks.

**Review by awgreene — APPROVED** [review]
/approve

**Pull request merged**